### ISSUE TITLE
Uncaught PHP Exception Doctrine\ORM\EntityNotFoundException: "Entity of type 'Mautic\LeadBundle\Entity\Lead' for IDs id(310794) was not found"

### ISSUE THREAD

**Issue by diletecsoftware** [mautic/mautic#8712]
[//]: # ( Invisible comment: 
IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII
Before you create the issue:
IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII
Search for similar report among other reported issues.
Learn how to troubleshoot at https://www.mautic.org/docs/en/tips/troubleshooting.html
Use drag&drop to attach images or other files )

## Bug Description
The more details the better...

| Q   | A
| --- | ---
| Mautic version | 2.15.3
| PHP version | 7.2.29
| Browser | Desktop and Mobile

### Steps to reproduce
1. Landing page.
The error happened with old and new Leads, they all came from a post on linkedin with the link directing to the landing page. When accessing the landing page, mautic reproduces error 500.

What was done:
  - Cache cleanup
  - Schema update
  - Lead purchase to try to locate some data entered in the wrong way
  - Lead update, registering data such as email, name and surname

Comments:
  - This happens with some leads
  - When accessing the same link from an anonymous tab the landing page works
  - The landing page contains HTML, CSS, JS, Bootstrap 4.0 and {form = idNamber}

Unfortunately I was only able to reproduce the error once on my smartphone, and no more. But I still receive email from a lead having a problem accessing the landing page, and looking at the log I see that it is the same error.
 
### Log errors
mautic.CRITICAL: Uncaught PHP Exception Doctrine\ORM\EntityNotFoundException: "Entity of type 'Mautic\LeadBundle\Entity\Lead' for IDs id(310794) was not found" at /home/dl/public_html/mkt/vendor/doctrine/orm/lib/Doctrine/ORM/EntityNotFoundException.php line 47 {"exception":"[object] (Doctrine\\ORM\\EntityNotFoundException(code: 0): Entity of type 'Mautic\\LeadBundle\\Entity\\Lead' for IDs id(310794) was not found at /home/ld/public_html/mkt/vendor/doctrine/orm/lib/Doctrine/ORM/EntityNotFoundException.php:47)"} []

mautic.CRITICAL: Uncaught PHP Exception Symfony\Component\Debug\Exception\FatalThrowableError: "Call to a member function getId() on null" at /home/dl/public_html/mkt/app/bundles/LeadBundle/Helper/ContactRequestHelper.php line 257 {"exception":"[object] (Symfony\\Component\\Debug\\Exception\\FatalThrowableError(code: 0): Call to a member function getId() on null at /home/ld/public_html/mkt/app/bundles/LeadBundle/Helper/ContactRequestHelper.php:257)"} []

mautic.CRITICAL: Uncaught PHP Exception Symfony\Component\Debug\Exception\FatalThrowableError: "Call to a member function getId() on null" at /home/dl/public_html/mkt/app/bundles/LeadBundle/Helper/ContactRequestHelper.php line 257 {"exception":"[object] (Symfony\\Component\\Debug\\Exception\\FatalThrowableError(code: 0): Call to a member function getId() on null at /home/ld/public_html/mkt/app/bundles/LeadBundle/Helper/ContactRequestHelper.php:257)"} []

[//]: # ( Invisible comment:
Please check for related errors in the latest log file in [mautic root]/app/log/ and/or the web server's logs and post them here. Be sure to remove sensitive information if applicable. )

**Comment by RCheesley** [comment]
Hi there, as we are only providing fixes for Mautic 3.x going forward, can you confirm if this continues in Mautic 3.0.1?
Thanks!

**Comment by RCheesley** [comment]
Closing due to a lack of recent activity. Please create a new issue and link to this one if necessary.

**Comment by diletecsoftware** [comment]
Hello. Hello
I installed Mautic 3.0.1 and I will check if the problem persists, if yes I will open a new issue.

**Comment by diletecsoftware** [comment]
Hello again.

**The error continues in version 3.0.2 of Mautic.**

[2020-08-17 13:20:03] mautic.CRITICAL: Uncaught PHP Exception Doctrine \ ORM \ EntityNotFoundException: "Entity of type 'Mautic \ LeadBundle \ Entity \ Lead' for IDs id (211796) was not found" at / home / mkt / public_html / mkt.com / vendor / doctrine / orm / lib / Doctrine / ORM / EntityNotFoundException.php line 47 {"exception": "[object] (Doctrine \\ ORM \\ EntityNotFoundException (code: 0): Entity of type 'Mautic \\ LeadBundle \\ Entity \\ Lead' for IDs id (211796) was not found at /home/mkt/public_html/mkt.com/vendor/doctrine/orm/lib/Doctrine/ORM/EntityNotFoundException. php: 47) "} []

**Comment by luguenth** [comment]
Does your Problem still persist? Or did you manage to solve it somehow?
It also occurs on one of our Instances, which is an upgraded Mautic 2 Instance now running Mautic 3

**Comment by danielwpsouza** [comment]
Hello how are you?

This problem persists, I know how to solve it, but I believe that this correction should be in the CORE of Mautic.

This happens when the "Lead" is deleted by the command below.

Run the command "php console mautic: maintenance: cleanup --dry-run --days-old = 90", after that the errors appear.

The fix is ​​to delete the "Orphan ID".

My correction:
I created a PHP file that fetches the lead_id in all tables in the database.
Then I make a check if these tables have a value in lead_id that does not exist in the leads table in the ID field.

So I delete this "ID" record in the tables that have this "orphan value".

Sorry, my English is not very good, yet. To help with clarifications I will leave here at least the queries:

 - "SELECT TABLE_NAME from INFORMATION_SCHEMA.COLUMNS WHERE COLUMN_NAME like '% lead_id%'"

 - "SELECT lc.lead_id FROM". $ Tabs-> TABLE_NAME. "AS lc LEFT OUTER JOIN` leads` AS ld ON lc.lead_id = ld.id WHERE ld.id IS NULL GROUP BY lc.lead_id "

 - "DELETE FROM $ tabs-> TABLE_NAME WHERE lead_id IN ($ inc)"

My current version is 3.3.3

**Comment by danielwpsouza** [comment]
@RCheesley 
Is there a console command to resolve this?
Some native action would be more interesting than my CRON in parallel.
